scripts/compatibility_scanner/compatibility_scanner.q

```diff
@@ -0,0 +1,239 @@
+  ];
+
+  -1"\nChecked ",string[count args`file]," .q script(s)";
+  -1"Total lines with possible incompatibilities: ",string count lines;
+
+  $[
+    (not ()~args[`csv])and 0<>count lines;writeToCsv[args`csv;checksDict;lines];
+    0~count lines;-1"WARN: Skipped writing to csv, no lines to output";
+    ()
+  ];
+
+  exit 0;
+ };
+
+getFilesFromRegex:{[regexList;dir]
+  if[0~count regexList;:()];
+
+  if[dir~();dir:first system"pwd"];
+  res:raze{system"find ",x," -regextype posix-extended -type f -regex '",y,"'"}[dir]each regexList;
+
+  :getFullPath ignoreNonQScripts res;
+ };
+
+ignoreNonQScripts:{[files]
+  res:files where files like "*[.]q";
+  if[count[res]<>count files;-1"WARN: Ignored files that are not .q scripts"];
+
+  :res;
+ };
+
+scanFile:{[file;checksDict]
+  res:system"printf '%s\\n' \"$(grep -nHE '",combineRegexList[value checksDict],"' '",file,"')\"";
+  
+  if[""~raze/[res];res:()];
+  if[0<count res;
+    -1"\n" sv res;
+  ];
+
+  :res;
+ };
+
+getDirFileList:{[dir]
+  :getFullPath system"echo \"$(find '",dir,"' -name '*\\.q')\"";
+ };
+
+parseArgs:{[]
+  if[$[0<>count .z.x;"--help" in .z.x;0b] or 0~count .z.x;showHelp[]];
+
+  args:(enlist[`]!enlist[::]),.Q.opt .z.x;
+
+  if[0h~type args`dir;args[`dir]:first args`dir];
+  if[10h<>type args`dir;args[`dir]:()];
+
+  if[0h<>type args`file;args[`file]:enlist args`file];
+  args[`file]:args[`file] where 10h=type each args`file;
+
+  if[0h<>type args`exclude;args[`exclude]:enlist args`exclude];
+  args[`exclude]:args[`exclude] where 10h=type each args`exclude;
+
+  args[`regex]:$[0<>count .z.x;"--regex" in .z.x;0b];
+
+  if[0h~type args`csv;args[`csv]:first args`csv];
+  if[10h<>type args`csv;args[`csv]:()];
+
+  :args;
+ };
+
+loadChecksDict:{[]
+  checksDict:enlist[`z_assignment]!enlist readAssignmentChecks ASSIGNMENT_CHECKS_TSV;
+  checksDict,:enlist[`system_command]!enlist readCommandsChecks COMMANDS_CHECKS_TSV;
+
+  if[DEBUG_SHOW_REGEX_PASSED;
+    -1"DEBUG: Regex pattern being passed:";
+    -1 combineRegexList value checksDict;
+  ];
+  
+  :checksDict;
+ };
+
+readCommandsChecks:{[file]
+  tbl:("**";enlist"\t") 0: hsym`$file;
+
+  :combineRegexList tbl`regex;
+ };
+
+readAssignmentChecks:{[file]
+  tbl:("**";enlist"\t") 0: hsym`$file;
+
+  :combineRegexList raze{
+    :enlist[y] cross x cross enlist z;
+  }[ZS_TO_CHECK]'[tbl`prefix;tbl`suffix];
+ };
+
+combineRegexList:{[checksList]
+  :-1 _ raze{"(",x,")|"}each checksList;
+ };
+
+filterExcluded:{[files;excludedFiles]
+  :files where not files in excludedFiles;
+ };
+
+writeToCsv:{[file;checksDict;lines]
+  file:getFullPath file;
+  if[not file like "*[.]csv";
+    -1"WARN: Adding '.csv' suffix to file '",file,"'";
+    file,:".csv";
+  ];
+
+  tbl:categoriseLines[checksDict;lines];
+
+  -1"Saving to file '",file,"' . . .";
+  @[{(hsym`$x) 0: csv 0: y;-1"Saved"}[file];tbl;
+    {[x;y] -2"ERROR: Could not save to '",x,"' due to:\n",y}[file]];
+ };
+
+categoriseLines:{[checksDict;lines]
+  -1"Categorising lines . . .";
```

Comment:
Maybe a log line along the lines of "Splitting output lines into categories...". A bit more descriptive about what is happening

Reply:
I agree yea, changed it now